src/Raven.Server/Documents/DocumentsNotifications.cs

```diff
@@ -48,7 +48,11 @@ public void RaiseNotifications(DocumentChangeNotification documentChangeNotifica
             OnDocumentChange?.Invoke(documentChangeNotification);
 
             foreach (var connection in Connections)
-                connection.Value.SendDocumentChanges(documentChangeNotification);
+            {
+                if (connection.Value.IsDisposed == false)
+                    connection.Value.SendDocumentChanges(documentChangeNotification);
```

Comment:
You still have a race condition here. Between the check for the dispose and the actual call.